pkg/util/metrics_helper.go

```diff
@@ -675,6 +675,25 @@ func (r *UserRegistries) Registries() []UserRegistry {
 	return out
 }
 
+// GetRegistryForUser returns currently active registry for given user, or nil, if there is no such registry.
+func (r *UserRegistries) GetRegistryForUser(user string) *prometheus.Registry {
+	r.regsMu.Lock()
+	defer r.regsMu.Unlock()
+
+	for idx := 0; idx < len(r.regs); idx++ {
+		if user != r.regs[idx].user {
+			continue
+		}
+
+		if r.regs[idx].reg == nil {
+			continue
```

Comment:
[nit] Can you add a comment to explain why? I had to look at the rest of the code to understand it. It's because the registry to set to nil on soft deletion, but the user may have a new registry registered.

Reply:
Yes, I was thinking about that use-case that you describe, but I just realized that we also set `user` to `""` on soft-deletion, so this if condition can never be true (unless one passes `""` parameter to the function). I will remove it.

Reply:
https://github.com/grafana/mimir/pull/3923/commits/863f9afc24f96a1a77d3b2fbee8ea0da57dedb2b